src/components/orders/OrderSurplusDisplay/index.tsx

```diff
@@ -39,7 +44,7 @@ export function OrderSurplusDisplay(props: Props): JSX.Element | null {
   // const usdAmount = '55.555'
 
   if (!surplusToken || surplusAmount.isZero()) {
-    return null
+    return {} as SurplusText
```

Comment:
You shouldn't need to cast if you have the right type.
Also, consumers would expect the type to be honoured.

Below on line 72 you check if the fields are undefined, which shouldn't possible according to the type.

I would either:
1. Keep the return `null`, to be clear there's no data
2. Make both fields optional `type SurplusText = { amount?: string; percentage?: string }`

In either case, you don't need the cast

Reply:
Thank you! It's Done!